eng/versioning/update_versions.py

```diff
@@ -80,6 +80,7 @@ def update_versions(update_type, version_map, ext_dep_map, target_file, skip_rea
                             repl_open, repl_thisline = False, False
 
                 if repl_thisline:
+                    newline = ''
```

Comment:
Why is `newline` an empty string?

Reply:
The script needs to declare the variable before use. We are only be declaring it in the local scope while using it in a global one. When I ran the script a few times, I saw that sometimes this newline was having an older value because of the last used scope and likely string not being GCed or not (my guess, but I am not aware of how things are GCed in python) so I wanted to declare the variable with a value for each iteration. A quick google search said null behaves a little different in python so, I used empty since 'if of an empty string' returns false which serves my purpose. If there is a better way to represent it in python happy to do it :) I do not know the language enough (started with it yesterday!), so happy to learn along the way.